app/uk/gov/hmrc/gform/gform/PrepopService.scala

```diff
@@ -17,26 +17,25 @@
 package uk.gov.hmrc.gform.gform
 
 import play.api.Logger
-import scala.util.control.NonFatal
-import uk.gov.hmrc.auth.core.retrieve.GGCredId
-import uk.gov.hmrc.gform.auth.models.{ AnonymousRetrievals, AuthenticatedRetrievals, MaterialisedRetrievals }
+import uk.gov.hmrc.gform.auth.models.MaterialisedRetrievals
+import uk.gov.hmrc.gform.models.mappings.{ HMRCOBTDSORG, IRCT, IRSA, NINO }
 import uk.gov.hmrc.gform.sharedmodel.formtemplate._
-import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext._
 import uk.gov.hmrc.http.HeaderCarrier
+import uk.gov.hmrc.play.http.logging.MdcLoggingExecutionContext._
+
+import scala.util.control.NonFatal
 
 object AuthContextPrepop {
   def values(value: AuthInfo, retrievals: MaterialisedRetrievals): String = value match {
     case GG                     => retrievals.ggCredId
-    case PayeNino               => retrievals.getTaxIdValue(None, "NINO")
```

Comment:
Can you use the field values in the corresponding ServiceNameAndTaxId values here instead of the Option values & literal strings?

Reply:
Same as above